### ISSUE TITLE
10$: Warn on conflict between parameter and var-declared variable maybe

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#896]
https://twitter.com/RReverser/status/1602820878907187200

The behavior of the following code differs between JS engines, so we should warn about it:
```javascript
async function initModule(foo) {
	var foo = foo || 2;
	console.log(foo);
}

initModule(1);
```

**Comment by strager** [comment]
This issue seems to be specific to `async` functions: https://bugs.webkit.org/show_bug.cgi?id=223533
